File: mycollab-web/src/main/scala/com/mycollab/module/project/event/MilestoneEvent.scala
package com.mycollab.module.project.event

import com.mycollab.events.ApplicationEvent

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
object MilestoneEvent {

  class GotoAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoRoadmap(source: AnyRef) extends ApplicationEvent(source, null) {}
  
  class GotoKanban(source: AnyRef) extends ApplicationEvent(source, null) {}

  class NewMilestoneAdded(source: AnyRef, data: Integer) extends ApplicationEvent(source, data) {}
  
  class MilestoneDeleted(source: AnyRef, data: Integer) extends ApplicationEvent(source, data) {}

}

File: mycollab-web/src/main/java/com/mycollab/module/project/event/MilestoneEvent.kt
package com.mycollab.module.project.event

import com.mycollab.events.ApplicationEvent

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class MilestoneEvent {
    class GotoAdd(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class GotoEdit(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class GotoList(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class GotoRead(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class GotoRoadmap(source: Any) : ApplicationEvent(source, null) 

    class GotoKanban(source: Any) : ApplicationEvent(source, null) 

    class NewMilestoneAdded(source: Any, data: Int) : ApplicationEvent(source, data) 

    class MilestoneDeleted(source: Any, data: Int) : ApplicationEvent(source, data) 
}